src/datasets/types.ts

```diff
@@ -0,0 +1,58 @@
+import { DataCube } from 'src/models/data-cube/data-cube.model';
+
+export interface Dataset  {
+  metas: Array<Meta>;
+  dataCube: DataCube;
+}
+
+export type Meta = TabbedChartsMeta | ChartMeta;
+
+export interface ComponentMeta<T extends string> {
+  type: T;
+  title: string;
+}
+
+export interface DataComponentMeta<T extends string, QueryT> extends ComponentMeta<T> {
+  query: QueryT;
+}
+
+export interface ContainerComponentMeta<T extends string, MetaT> extends ComponentMeta<T> {

```

Comment:
`MetaT extends (Meta | Meta[])`?

I think it would be cleaner to have two separate interfaces for singular and plural metas respectively.

Reply:
I assume `ContainerComponentMeta` can take any object composed of structure information and `Meta`s. For example, a `GridChartsMeta` may take a 2-dimensional array of `Meta`.

However, I don't know if there will be other container components in the future. It may be overengineered to take it into consideration.

Reply:
I was mostly concerned of naming. A container of a single meta should have `meta`, that of multiple metas should have `metas`, that of `Map<Meta>` should have `metaMap`, etc.

With that being said, is there a reason that we need `ContainerComponentMeta`? I think it's okay to not have a common parent interface unless we need to group them for their common behavior.

Reply:
Alright, I agree there should not exist a `ContainerComponentMeta` as parent. I will remove it, thanks!